internal/manager/config/config.go

```diff
@@ -640,6 +641,14 @@ func (i *Instance) GetVideoFileNamingAlgorithm() models.HashAlgorithm {
 }
 
 func (i *Instance) GetGalleryCoverRegex() string {
+	var regexString = i.getString(GalleryCoverRegex)
+
+	_, err := regexp.Compile(regexString)
+	if err != nil {
+		log.Warn(fmt.Sprintf("Gallery cover regex '%v' invalid, reverting to default.", regexString))
```

Comment:
I haven't testing anything myself but I just noticed that the log function used here is technically from the wrong place (plugins), and while I think those just delegate to the normal logging functions you should probably just be using `logger.Warnf(...)` directly.

And also while I'm here I can't help but do some very minor nitpicking: the value from `i.getString` is already in the `regexString` variable so that can be returned directly instead of retrieving the value again.

Regarding the validation, any overhead is probably very minimal unless the regex is really complicated, and as mentioned the check can be moved to the saving section after this is added to the UI anyway.

Reply:
My apologies, I should probably have been more clear before - `logger.Warnf` does Printf formatting itself so the `fmt.Sprintf` is redundant - ie it can just be `logger.Warnf("Gallery cover regex '%v' invalid, reverting to default.", regexString)`.